src/_arraykit.c

```diff
@@ -147,6 +148,10 @@ AK_ResolveDTypeIter(PyObject *dtypes)
         }
     }
     Py_DECREF(iterator);
+    if (!resolved) {
+        // this could happen if this function gets an empty tuple
+        PyErr_SetString(PyExc_ValueError, "iterator passed to resolve dtypes is empty");
```

Comment:
Sorry, one more minor nit:

```suggestion
        PyErr_SetString(PyExc_ValueError, "iterable passed to resolve dtypes is empty");
```